### ISSUE TITLE
Passing a start solution via HiGHS C# API

### ISSUE THREAD

**Issue by jannicklange** [ERGO-Code/HiGHS#972]
Hi!

I'm trying to access the HiGHS solver through the [C# interface](https://github.com/ERGO-Code/HiGHS/blob/master/src/interfaces/highs_csharp_api.cs).

I found the [Highs_setSolution](https://github.com/ERGO-Code/HiGHS/blob/c0e1182bd350d4ecfc4600aee3e7ff71aacae6cd/src/interfaces/highs_c_api.h) in your C api, and tried calling it analogous to the other API calls in the C# interface:

```
[DllImport(highslibname)]
private static extern int Highs_setSolution(
        IntPtr highs,
        double[] col_value,
        double[] row_value,
        double[] col_dual,
        double[] row_dual);

public HighsStatus SetStartSolution(HighsSolution solution)
{
    return (HighsStatus)HighsLpSolver.Highs_setSolution(
        this.highs,
        solution.colvalue,
        solution.coldual,
        solution.rowvalue,
        solution.rowdual);
}
```
The extern call fails, stating that the `Highs_setSolution` method cannot be found on my version of the `libhighs.dll`. I'm using the DLL from the [HiGHS.v1.2.2.x86_64-w64-mingw32-cxx11.tar.gz](https://github.com/JuliaBinaryWrappers/HiGHS_jll.jl/releases) package.

According to this [commit](https://github.com/ERGO-Code/HiGHS/commit/0c5f3712f64c6a6ce799dbef7f039beef32f1928) and the `1.2.2` Tag, I'd expect the method to exist. The PR was merged on 2022-04-14, the [release tag](https://github.com/ERGO-Code/HiGHS/releases/tag/v1.2.2) was created on 2022-04-18. However, I noticed that you linked the zipped source code in archives that were created on 2022-04-08. 

Is it possible, that the PR #508 is not part of the actual deployed version 1.2.2? Or do I need to change my approach on extending the C# interface?
Note: I already added a `getInfo` method to the C# interface, which returns the `"mip_gap", "mip_dual_bound", ...`-HighsInfos, by using an equivalent approach as described above (i.e. relying on extern calls to the `highs_c_api`).

Thanks,
Jannick

**Comment by jajhall** [comment]
Indeed, `Highs_setSolution` isn't in the C api for `1.2.2`. It's been added since, and will be available in the new release - which is overdue. `1.2.2` is dated 8 April 2022, so #508 won't be part of the release. I guess that after creating `1.2.2` we should have incremented the version counter within `master`. I appreciate that our version control is of nothing like the standard of our solvers but, with the size of team and experience that we've got, software engineering and solver performance is our priority.

The C# interface hasn't been updated for a long time because we're not C# users to the extent that I don't know how to run HiGHS from a C# driver such as [call_highs_from_csharp.cs](https://github.com/ERGO-Code/HiGHS/blob/master/examples/call_highs_from_csharp.cs). Hence the C# interface will be short of many methods that are in the C api. I've thought of creating them by pattern-matching, but that seems irresponsible without a way of checking whether they even compile. _It would be helpful if you could create a PR of the changes that you've made._ That said, your name `SetStartSolution` doesn't match the naming convention whereby method `foo` in the C++ class is called via method `Highs_foo` in the C API that, in turn is called `foo` in the C# API.

**Comment by jannicklange** [comment]
Thanks for the clarification. I'll be looking forward to the next release :)
I created a PR for extending the C# API. If we add further C# functionality in the future, I'll try to keep the public version synched.

**Comment by jajhall** [comment]
I assume that this issue is closed by #974
